saleor/discount/utils.py

```diff
@@ -591,6 +638,7 @@ def _set_checkout_base_prices(checkout_info, lines_info):
 
 
 def _clear_checkout_discount(checkout_info, save):
+    # TODO: remove gift if exist - it should also remove CheckoutLineDiscount
```

Comment:
Should it be fixed in a separate PR?

Reply:
Yes, as described in the PR description. Will be fixed in https://github.com/saleor/saleor/issues/15323. PR is waiting: https://github.com/saleor/saleor/pull/15335